docs/CONTRIBUTING.md

```diff
@@ -223,6 +223,74 @@ unstable features:
 cargo +nightly fmt
 ```
 
+### Code Coverage
+
+We appreciate well-tested code, so feel free to add tests when you can.
+
+To generate code coverage using the cypress tests, follow these steps:
+
+0. Make sure `grcov` and `llvm-tools-preview` are installed
+
+   ```shell
+   rustup install llvm-tools-preview
+   cargo install grcov
+   ```
+
+1. Build the project with the `-Cinstrument-coverage` flag:
+
+   ```shell
+   RUSTFLAGS="-Cinstrument-coverage" cargo build --bin=router --package=router
+   ```
+
+   Several `.profraw` files will be generated. (Due to the execution of build scripts)
+
+2. Execute the binary:
+
+   ```shell
+   LLVM_PROFILE_FILE="coverage.profraw" target/debug/router
+   ```
+
+3. Open a separate terminal tab and run the cypress tests, following the [README]
```

Comment:
Nit:

```suggestion
3. Open a separate terminal tab and run the cypress tests, following the [README][cypress-v2-readme]
```

And update the link to be:

```markdown
[cypress-v2-readme]: /cypress-tests-v2/README.md
```

@pixincreate Should we consider linking to the v1 README file instead? 
Should we also remove all content from `cypress-tests-v2/README.md`, and update it to suggest referring to the `cypress-tests/README.md` file, to ease maintenance?

Reply:
the thing is that, in v2, entire format / architecture is different from v1. and removing readme from might create some chaos imo.

i'm also planning to slowly start moving to `ts` in the coming days. 

i prefer to keep it, at least for now.